### PR TITLE
docs: release note remove warning about binaries

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR remove the warning section of release note page and add a note to new Binary Policy page to explain more deeply what `<next-version-number>~pre` is with example.

OP #5373

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

File: docs/manuals/source/Appendix/ReleaseNotes.rst [hunk 1]
```diff
@@ -5,12 +5,11 @@ Release Notes
...
 
-.. warning::
-
-      While all the source code is published on `GitHub <https://github.com/bareos/bareos/>`_, the releases of packages on https://download.bareos.org/ is limited to the initial versions of a major release. Later maintenance releases are only published on https://download.bareos.com/.
+This information is also available as ``CHANGELOG.md`` in the corresponding branch of the `Bareos GitHub project`_
 
+.. note::
 
-This information is also available as ``CHANGELOG.md`` in the corresponding branch of the `Bareos GitHub project`_
+   You can consult our :ref:`section-BareosBinaryReleasePolicy` section to get more information about version available.
```
Comment (joergsteffens) [inline docs/manuals/source/Appendix/ReleaseNotes.rst:12]:
```suggestion
   See :ref:`section-BareosBinaryReleasePolicy` for more information about available versions.
```

File: docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst [hunk 2]
```diff
@@ -164,11 +164,11 @@ Not all packages are required to run Bareos.
 
 .. _section-BareosBinaryReleasePolicy:
 
-.. _section-BareosRepositoryTypes:
-
 Bareos Binary Release Policy
 ----------------------------
 
+.. _section-BareosRepositoryTypes:
```
Comment (joergsteffens) [inline docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:170]: Why have moved this below the section title? IMHO it belongs above the title. Multiple section labels are not a problem.

File: docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst [hunk 3]
```diff
@@ -180,13 +180,13 @@ There are different types of Bareos binaries:
...
       * Packages may be marked as pre-releases (``<next-version-number>~pre``) and are published after passing an automated testing process.
-      * When a new Bareos major version gets released, the version in this repository will also change to the new version.
+      * When a new Bareos major version gets released, this repository will also change to the new current version.
 
    * Latest build of the Bareos master branch at https://download.bareos.org/next/
 
       * Only the latest build is available.
       * Packages are marked as pre-releases (``<next-major-version-number>.0.0~pre``) and are only published after passing an automated testing process.
-      * When a new Bareos major version gets released, the version in this repository will also change to the new version.
+      * When a new Bareos major version start to be developed, this repository will also change to the new next version.
```
Comment (joergsteffens) [inline docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:189]:
```suggestion
      * When the development of a new Bareos major version starts, this repository will also change to the new **next** version.
```

File: docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst [hunk 4]
```diff
@@ -180,13 +180,13 @@ There are different types of Bareos binaries:
 
       * Only the latest build is available.
       * Packages may be marked as pre-releases (``<next-version-number>~pre``) and are published after passing an automated testing process.
-      * When a new Bareos major version gets released, the version in this repository will also change to the new version.
+      * When a new Bareos major version gets released, this repository will also change to the new current version.
```
Comment (joergsteffens) [inline docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:183]:
```suggestion
      * When a new Bareos major version gets released, this repository will also change to the new **current** version.
```

File: docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst [hunk 5]
```diff
@@ -198,6 +198,13 @@ There are different types of Bareos binaries:
 
 The software in both types of repositories is based on the same source code freely available in https://github.com/bareos/bareos/. There are no hidden nor open core components.
 
+.. note::
+
+   About (``<next-version-number>~pre``) for example if we have released 22.0.1 and some code has been backported in the meantime to the current (22) version,
+   newer packages with a 22.0.2~pre version will get published. It means latest stable current 22.0.1 release + newer backported feature that will appear in 22.0.2.
```
Comment (joergsteffens) [inline docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:204]:
```suggestion
   About ``<next-version-number>~pre`` version numbers:
   
   Assume the current major release is ``22`` and the latest stable release is ``22.0.1``.
   As soon as some code gets backported to Bareos 22,
   packages with ``22.0.2~pre`` version will get published.
   It means the latest stable current ``22.0.1`` release + newer backported feature that will appear in ``22.0.2``.
```

Does this note not only belong to the https://download.bareos.org/current/ part?

**Review by joergsteffens — CHANGES_REQUESTED** [review]
This is an improvement, but I'd formulate some things differently.

File: docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst (hunk 5 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:204]: yes only d.b.o/current is concerned, I've tried to bring the note just after but it make the split between the two type of repo less evident.

File: docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst (hunk 2 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:170]: Don't know what was not working, restested and it works. Would you like to see this title going back before the section?

Comment (joergsteffens) [reply docs/manuals/source/IntroductionAndTutorial/WhatIsBareos.rst:170]: Either move it above the section title (as this ref is also called "section-") or remove it completely, as it is not used anymore anyway.

**Review by joergsteffens — APPROVED** [review]
I added the CHANGELOG. Fixing the section ref would be nice, but otherwise approved the PR.

**Comment by bruno-at-bareos** [comment]
Done.

**Pull request merged**